pkg/activator/throttler.go

```diff
@@ -132,14 +146,29 @@ func (t *Throttler) Try(rev RevisionID, function func()) error {
 	return nil
 }
 
+func (t *Throttler) activatorEndpointsUpdated(newObj interface{}) {
+	endpoints := newObj.(*corev1.Endpoints)
+	count := resources.ReadyAddressCount(endpoints)
+	atomic.StoreInt32(&t.numActivators, int32(count))
+	t.updateAllBreakerCapacity()
+}
+
+func (t *Throttler) activatorCount() int {
+	activatorCount := int(atomic.LoadInt32(&t.numActivators))
+	if activatorCount > 0 {
+		return activatorCount
+	}
+	return 1
+}
+
 // This method updates Breaker's concurrency.
 func (t *Throttler) updateCapacity(breaker *queue.Breaker, cc, size int) (err error) {
 	targetCapacity := cc * size
 	if size > 0 && (cc == 0 || targetCapacity > t.breakerParams.MaxConcurrency) {
 		// The concurrency is unlimited, thus hand out as many tokens as we can in this breaker.
 		targetCapacity = t.breakerParams.MaxConcurrency
 	}
-
+	targetCapacity = targetCapacity / t.activatorCount()
```

Comment:
What happens here in the case of concurrency: 1, size: 1 where I am running two activators - will this return .5 forever which gets rounded down to 0?

Reply:
Probably we should enforce min of 1.

Reply:
Or even take `Ceil` vs `Floor`? That'd fix my comment on "losing" capacity as well. With multiple activators we're prone to overscheduling anyway, might as well make some of the edge cases a bit more prone.

Reply:
I think in we want to take the `floor` but min of 1, as it'll be queued up before forwarding to the pods if it sees the pod will be overloaded. In the special case where you less cc than activators, you may end up with very few dropped requests.

Reply:
ISTM we'll want to get to a place where we are oversubscribing in activator but watching for brackpressure. That'd be a pretty significant change though since right now probing happens in activator handler. For now I have a hard time arguing one way or the other - if we undersubscribe its not great as we already have queue depth issues which I expect this could exacerbate, if we oversubscribe we don't have a good way to notice backpressure ATM...

Reply:
I also think that we should not probe revision from all the requests the activator receives. I think the work you do to move it to the throttler should help (and we can leverage the async prober component I've built for the KPA).